corpus/pages/views.py

```diff
@@ -26,9 +28,27 @@ def about_us(request):
         },
     )
 
+
 def impulse(request):
 
     return render(
         request,
         "pages/impulse.html",
-    )
\ No newline at end of file
+    )
+
+
+def sig(request, sig_name):
+    sig_data = get_object_or_404(SIG, name=sig_name)
+
+    # Retrieve the related society details using the SIG instance
+    society_linked_to_sig = Society.objects.filter(sig=sig_data.id)
+
+    return render(
+        request,
+        "pages/sig.html",
+        {
+            "sig": sig_data,
+            # Pass the society details to the SIG
+            "society_linked_to_sig": society_linked_to_sig,
+        },
```

Comment:
NIT: Create an `args` dictionary differently, and pass that to the function. Lets keep functions as minimal and one-liner as possible. Refer to https://github.com/IEEE-NITK/corpus/blob/main/corpus/embedathon/views.py#L275 as reference.